convert-hf-to-gguf.py

```diff
@@ -1517,6 +1517,119 @@ def write_tensors(self):
             raise ValueError(f"Unprocessed experts: {experts.keys()}")
 
 
+@Model.register("ArcticForCausalLM")
+class ArcticModel(Model):
+    model_arch = gguf.MODEL_ARCH.ARCTIC
+
+    def set_vocab(self):
+        self._set_vocab_llama_hf()
+
+    def set_gguf_parameters(self):
+        super().set_gguf_parameters()
+        hparams = self.hparams
+        self.gguf_writer.add_vocab_size(hparams["vocab_size"])
+        self.gguf_writer.add_rope_dimension_count(hparams["hidden_size"] // hparams["num_attention_heads"])
+
+    # Same as super class, but permuting q_proj, k_proj
+    def write_tensors(self):
```

Comment:
This can be simplified with the changes from #7075 by overriding `modify_tensors` instead of `write_tensors`.

Reply:
@compilade I applied changes from #7075 and #6511 to the ArcticModel in convert-hf.